pyccel/naming/cnameclashchecker.py

```diff
@@ -71,17 +71,33 @@ def get_collisionless_name(self, name, symbols):
             The suggested name.
         symbols : set
             Symbols which should be considered as collisions.
+        prefix : str
+            The prefix that may be added to the name to provide context information.
+        context : str
+            The context where the name will be used.
+        parent_context : str
+            The type of the scope where the object with this name will be saved.
 
         Returns
         -------
         str
             A new name which is collision free.
         """
-        if name in ('__init__', '__del__'):
-            return name
+        assert context in ('module', 'function', 'class', 'variable', 'wrapper')
+        assert parent_context in ('module', 'function', 'class', 'loop', 'program')
+        if context == 'wrapper':
+            # wrapper names are based off names which already have prefixes so there is no
+            # need to add more
+            return self._get_collisionless_name(name, symbols)
+        if name == '__init__':
+            name = 'create'
+        if name == '__del__':
+            name = 'drop'
```

Comment:
Why not `init` and `free`?

Reply:
`drop` was chosen to match what is used in STC. STC uses `drop` for destructors and matching this makes it simpler to call the destructor of the class when creating a list of classes or similar.

E.g. here are the docs for list: https://github.com/stclib/STC/blob/main/docs/list_api.md

For a list of classes we need to declare (plus a copy operator called MyClass_clone):
```c
#define i_keyclass MyClass
```
whereas if we call this method `free` we would need:
```c
#define i_key MyClass
#define i_keydrop MyClass_free
#define i_keyclone MyClass_clone
```

Reply:
I thought `create` was also chosen to match STC but either I got mixed up or this is no STC changed this. I have changed to `init` : f53009059

Reply:
Or maybe I just copied from Fortran for create